### PR TITLE
bots: Add test fixture files for bots.

### PR DESCRIPTION
**Pull request by abhijeetkaur**

### CODE REVIEW

**Comment by abhijeetkaur** [comment]
Added test-fixtures for all the bots with already existing tests. 

Adding a separate json for request/response behavior of the bot encourages people to add more test cases as one does not need to look into the code to see where to add the test samples. Folks can directly open up the test-fixture files and add tests.

File: zulip_bots/zulip_bots/bots/followup/fixtures/test_send_reply.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "expectations":[
+        {
+            "request":"",
+            "response":"Please specify the message you want to send to followup stream after @mention-bot"
+        }
+    ]
+}
```
Comment (aero31aero) [inline zulip_bots/zulip_bots/bots/followup/fixtures/test_send_reply.json]: Add a new line at the end. Ideally the linter should have caught this.

Comment (timabbott) [inline zulip_bots/zulip_bots/bots/followup/fixtures/test_send_reply.json]: I think the linter is probably not running against the JSON files?  We do that in the server repository, but maybe that bit of config wasn't ported over.

**Comment by timabbott** [comment]
@abhijeetkaur the file names for the tests here could be better.  E.g. for `followup_bot`, you have `test_send_message` and `test_send_reply`, but I think those should be `test_successful_request` and `test_empty_request`, since the fact that it's using `send_reply` is unimportant; what's important is that the input is invalid.

**Comment by timabbott** [comment]
I'm also confused by the test fixture strategy here.  What I had been imagining is that we would be adding test fixtures that mock the return values of the network requests (using `mock_http_conversation`, while keeping the actual testing code mostly unchanged.  This appears to have changed the testing code to be split between code and fixtures files, while not actually solving the problem that the tests won't work without Internet.

**Comment by abhijeetkaur** [comment]
@timabbott Oh, I wasn't aware of the primary objective.Well, I have mostly worked on the files not using internet. 
So, we intend to add test-fixtures for only those bots that use the internet, right?

**Comment by timabbott** [comment]
@abhijeetkaur I'm going to go ahead and close this; thanks again for your work on this and also your more recent work on adding fixtures for all the network connections in our bot tests.  The tests are a lot faster now, and my hope is that with a bit more work we can have a test suite we can be proud of.

**Pull request closed without merging**